src/common.rs

```diff
@@ -53,12 +53,14 @@ fn metadata_command(args: MetadataArgs) -> MetadataCommand {
     command
 }
 
-pub fn sourced_dependencies(metadata_args: MetadataArgs) -> Vec<SourcedPackage> {
+pub fn sourced_dependencies(
+    metadata_args: MetadataArgs,
+) -> Result<Vec<SourcedPackage>, anyhow::Error> {
     let command = metadata_command(metadata_args);
     let meta = match command.exec() {
         Ok(v) => v,
-        Err(cargo_metadata::Error::CargoMetadata { stderr: e }) => err_exit(&e),
-        Err(err) => err_exit(format!("Failed to fetch crate metadata!\n  {}", err).as_str()),
+        Err(cargo_metadata::Error::CargoMetadata { stderr: e }) => return Err(anyhow!(e)),
+        Err(err) => return Err(anyhow!("Failed to fetch crate metadata!\n  {}", err)),
```

Comment:
This could be better accomplished by using the [`Context` mechanism in `anyhow`](https://docs.rs/anyhow/1.0.57/anyhow/trait.Context.html):

```suggestion
        Err(err) => err.context("Failed to fetch crate metadata")?;
```

Please convert similar patterns to use `context`.